[DOCS] Adds peak_model_bytes and assignment_memory_basis to GET model snapshot API docs (#75413)
elastic/elasticsearch: docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc

`model_bytes_memory_limit`:::
(long) The upper limit for memory usage, checked on increasing values.

`peak_model_bytes`:::

> (long) The maximum memory usage of the model.

Reviewer: This sounds a bit like the maximum that **can** be used. How about we rephrase to `The highest recorded value for the model memory usage`?

Author: ```suggestion
(long) The highest recorded value for the model memory usage.
```
